test/langtools/tools/javac/records/RecordsBinaryCompatibilityTests.java

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
```

Comment:
this code is not strictly related to the patch but this test is just basically testing the changes Chapter 13 int the record spec. I'm OK moving this to another PR if needed

Reply:
not really but it is true that it is not strictly related to the patch, will create another issue for it